wgpu-core/src/storage.rs

```diff
@@ -44,26 +44,48 @@ pub(crate) struct InvalidId;
 /// values, so you should use an id allocator like `IdentityManager`
 /// that keeps the index values dense and close to zero.
 #[derive(Debug)]
-pub struct Storage<T, I: id::TypedId> {
+pub struct Storage<T, I: id::TypedId>
+where
+    T: Resource<I>,
+{
```

Comment:
It's a bit nicer to have the bounds all in one place, not split:

```suggestion
pub struct Storage<T, I>
where
    T: Resource<I>,
    I: id::TypedId,
```